### PR TITLE
dist/tools/buildsystem_sanity_check: add an export variable check

### PR DESCRIPTION
**Pull request by cladmi**
### Contribution description

Check that some variables are not exported in the build system.
This should track variables that managed to not be exported anymore so
that they do not reappear in a BSP.

It is not a whitelist but just a way to keep things cleaned in the
future.

### Testing procedure

No error when running the script as will CI check in `static-tests`.
No error when running `shellcheck` on the script.

Applying this diff will report a lot of errors as they are still exported.

```
diff --git a/dist/tools/buildsystem_sanity_check/check.sh b/dist/tools/buildsystem_sanity_check/check.sh
index e18ca6776..bd4071fd8 100755
--- a/dist/tools/buildsystem_sanity_check/check.sh
+++ b/dist/tools/buildsystem_sanity_check/check.sh
@@ -50,6 +50,8 @@ check_not_parsing_features() {
 UNEXPORTED_VARIABLES=()
 UNEXPORTED_VARIABLES+=('FLASHFILE')
 UNEXPORTED_VARIABLES+=('TERMPROG' 'TERMFLAGS')
+UNEXPORTED_VARIABLES+=('FFLAGS')
+UNEXPORTED_VARIABLES+=('PORT[ ?=]' 'PORT$')
 check_not_exporting_variables() {
     local patterns=()
     local pathspec=()
```

I used `PORT` here on purpose as it may need to be more complicated to not match `PORT_LINUX` too. Even if `PORT_LINUX` should not be exported anyway but another `PORTSOMETHING` may need to.

### Issues/PRs references

Tracking: remove harmful use of `export` in make and immediate evaluation #10850 
Useful for https://github.com/RIOT-OS/RIOT/pull/10440

### CODE REVIEW

File: dist/tools/buildsystem_sanity_check/check.sh [hunk 1]
```diff
@@ -41,11 +41,39 @@ check_not_parsing_features() {
...
+# It just keep things not exported anymore in the future.
+UNEXPORTED_VARIABLES=()
+UNEXPORTED_VARIABLES+=('FLASHFILE')
+UNEXPORTED_VARIABLES+=('TERMPROG' 'TERMFLAGS')
+check_not_exporting_variables() {
+    local patterns=()
+    local pathspec=()
+
+    for variable in "${UNEXPORTED_VARIABLES[@]}"; do
+        patterns+=(-e "export ${variable}")
```
Comment (jcarrano) [inline dist/tools/buildsystem_sanity_check/check.sh]:
```suggestion
        patterns+=(-e "export[[:blank:]]\+${variable}")
```

**Comment by cladmi** [comment]
I pushed force a line cleanup that stayed in my repository. Putting a fixup would still break the static tests that I would like to show they are passing.

File: dist/tools/buildsystem_sanity_check/check.sh [hunk 2]
```diff
@@ -41,11 +41,38 @@ check_not_parsing_features() {
...
+    done
+
+    pathspec+=('*')
+
+    # Ignore `makefiles/vars.inc.mk` as it currently is the only place that
+    # should export common variables.
+    pathspec+=(':!makefiles/vars.inc.mk')
+
+    git -C "${RIOTBASE}" grep "${patterns[@]}" -- "${pathspec[@]}"
```
Comment (jcarrano) [inline dist/tools/buildsystem_sanity_check/check.sh]:
```suggestion
    (set -x; git -C "${RIOTBASE}" grep "${patterns[@]}" -- "${pathspec[@]}")
```

Comment (cladmi) [reply dist/tools/buildsystem_sanity_check/check.sh]: I do not want to display it at all. It was just a stupid debug line remaining.

File: dist/tools/buildsystem_sanity_check/check.sh (hunk 1 above)
Comment (cladmi) [reply dist/tools/buildsystem_sanity_check/check.sh]: I indeed forgot the `\+` and including all whitespaces is safer.
There should not be tabs but does not hurt.

It should be supported by every versions as it is `posix` regex.

**Comment by jcarrano** [comment]
Squash and we're done.

**Comment by cladmi** [comment]
Squashed :)

**Comment by cladmi** [comment]
Thanks for the review. Now we can start removing exported variables more easily :)

**Pull request merged**